app/gui2/shared/ast/tree.ts

```diff
@@ -1164,22 +1165,30 @@ export interface MutableImport extends Import, MutableAst {
 }
 applyMixins(MutableImport, [MutableAst])
 
-const mapping: Record<string, string> = {
-  '\b': '\\b',
-  '\f': '\\f',
-  '\n': '\\n',
-  '\r': '\\r',
-  '\t': '\\t',
-  '\v': '\\v',
-  '"': '\\"',
-  "'": "\\'",
-  '`': '``',
-}
+const escapePairs = [
+  ['\0', '\\0'],
+  ['\b', '\\b'],
+  ['\f', '\\f'],
+  ['\n', '\\n'],
+  ['\r', '\\r'],
+  ['\t', '\\t'],
+  ['\v', '\\v'],
+  ["'", "\\'"],
+  ['`', '``'],
+] as const
+
+const escapeMapping = Object.fromEntries(escapePairs)
+const unescapeMapping = Object.fromEntries(escapePairs.map(([k, v]) => [v, k]))
 
 /** Escape a string so it can be safely spliced into an interpolated (`''`) Enso string.
- * NOT USABLE to insert into raw strings. Does not include quotes. */
-function escape(string: string) {
-  return string.replace(/[\0\b\f\n\r\t\v"'`]/g, (match) => mapping[match]!)
+ * NOT USABLE to insert into raw strings. Does not include double-quotes (`"`). */
+export function escapeInterpolation(string: string) {
+  return string.replace(/[\0\b\f\n\r\t\v'`]/g, (match) => escapeMapping[match]!)
+}
+
+/** Interpret all escaped characters from an interpolated (`''`) Enso string. */
+export function applyInterpolation(string: string) {
+  return string.replace(/\\[0bfnrtv']|``/g, (match) => unescapeMapping[match]!)
```

Comment:
I wouldn't know what this function is doing from this name. I'd rather expect some replacing parts in backticks.